modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Swift3Codegen.java

```diff
@@ -546,4 +561,47 @@ public String escapeUnsafeCharacters(String input) {
         return input.replace("*/", "*_/").replace("/*", "/_*");
     }
 
+    private static CodegenModel reconcileProperties(CodegenModel codegenModel, CodegenModel parentCodegenModel) {
+        // To support inheritgance in this generator, we will analyze
+        // the parent and child models, look for properties that match, and remove
+        // them from the child models and leave them in the parent.
+        // Because the child models extend the parents, the properties will be available via the parent.
+
+        // Get the properties for the parent and child models
+        final List<CodegenProperty> parentModelCodegenProperties = parentCodegenModel.vars;
+        List<CodegenProperty> codegenProperties = codegenModel.vars;
+        codegenModel.allVars = new ArrayList<CodegenProperty>(codegenProperties);
+        codegenModel.parentVars = parentCodegenModel.allVars;
+
+        // Iterate over all of the parent model properties
+        boolean removedChildProperty = false;
+
+        for (CodegenProperty parentModelCodegenProperty : parentModelCodegenProperties) {
+          // Now that we have found a prop in the parent class,
+          // and search the child class for the same prop.
+          Iterator<CodegenProperty> iterator = codegenProperties.iterator();
+          while (iterator.hasNext()) {
+              CodegenProperty codegenProperty = iterator.next();
+              if (codegenProperty.equals(parentModelCodegenProperty)) {
+                  // We found an enum in the child class that is
+                  // a duplicate of the one in the parent, so remove it.
+                  iterator.remove();
+                  removedChildProperty = true;
+              }
+          }
+        }
+
+        if(removedChildProperty) {
+            // If we removed an entry from this model's vars, we need to ensure hasMore is updated
+            int count = 0, numVars = codegenProperties.size();
+            for(CodegenProperty codegenProperty : codegenProperties) {
+                count += 1;
+                codegenProperty.hasMore = (count < numVars) ? true : null;
```

Comment:
I took this from other code generators.
It looks like CodegenProperty's properties are all initialized with null, and kept that way if not set. https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/CodegenProperty.java

See:
* https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaCodegen.java#L841
* https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavascriptClientCodegen.java#L999